### DISCUSSION TITLE
Issue while creating a Federation (interactive-api)⚙

### DISCUSSION THREAD

**Discussion by daviddelriod** [securefederatedai/openfederatedlearning#755] [Q&A] [answered]
Hi there,

I've been trying to simulate my _own FL experiment_ but I am encountering an error while **creating** the **federation**. My Federation cell looks like this: 
```
client_id = 'api'
cert_dir = 'cert'
director_node_fqdn = 'localhost'
director_port = '50051'

federation = Federation(
    client_id=client_id,
    director_node_fqdn=director_node_fqdn,
    director_port=director_port, 
    tls=False
)
```
And I get this **error** while executing the cell:
```
_InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "failed to connect to all addresses"
	debug_error_string = "{"created":"@1677873513.770000000","description":"Failed to pick subchannel","file":"src/core/ext/filters/client_channel/client_channel.cc","file_line":4143,"referenced_errors":[{"created":"@1677873513.770000000","description":"failed to connect to all addresses","file":"src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc","file_line":398,"grpc_status":14}]}"
>
```
My **envoy** and **director** configuration is almost the same as the one of the [TinyImage tutorial](https://github.com/securefederatedai/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_TinyImageNet). I also provide my [repo's link](https://github.com/daviddelriod/FL-SecFlow) (the notebooks placed in the workspace folder are irrelevant except the federation cell).

I attach a screenshot of my anaconda terminals and how I launch them.
![commands](https://user-images.githubusercontent.com/64251001/222825783-9e600095-902f-4e98-af88-bf87c2815d07.png)

I am pretty sure that the problem is related with how I launch the scripts or maybe something related with terminal (although I am using a conda virtual environment called TFG-FL with the OpenFL library). Any kind of help would be appreciated. Thanks!

**Comment by psfoley** [comment]
Hi @daviddelriod - your configuration looks fine to me. Do you have any proxies set up on the system? If so, I would recommend setting `export no_proxy=localhost` for all of the terminals you shared above. Was the envoy able to connect to the director? There should be a status message that appears in both terminals.

**Reply by daviddelriod** [reply]
I have noticed that when I run either of the .sh files, it opens a cmd window for 3 seconds and shows the message _bash **fx** command not found_.  This clearly indicates that the nodes are not connected. 

I don't know why this happens because, as I have shown, I run the script inside a conda environment but it seems that the .sh files are executed in another external cmd. How should I execute the scripts? (Note that I have got the same error while trying this in the Virtual Studio powershell with the environment activated, in the anaconda prompt and in the windows cmd too).

**Comment by daviddelriod** [comment, marked as answer]
Issue solved! 🙌 

The problem relied on the conda environment that I was using. I had the library working in visual studio but it was not installed in conda so I did `conda install git pip` and then `pip install git+https://github.com/securefederatedai/openfl` and the director started working (the envoys are struggling but for different errors😁).
